bbc/simorgh
# WSTEAM1-814: Adds Inline Script for Opera Mini ATI Beacon Request

Resolves JIRA [WSTEAM1-814](https://jira.dev.bbc.co.uk/browse/WSTEAM1-814)

Overall changes
======
Adds a function that uses the `XMLHttpRequest` api to send the ATI beacon for page views on Opera Mini, via an Inline script added via React Helmet on server side render.

Code changes
======

- Adds the `sendBeaconOperaMiniScript` function which checks if browser is opera mini and does a GET request to log the ATI page view, but exported as a string that can be inlined with calling code.
- Adds the inline script via `sendBeaconOperaMiniScript` using React Helmet in `CanonicalATIAnalytics`.
- Adds a check for the `sendBeacon()` function to only fire when browser is not opera mini.
- Converts the `isOperaProxy()` from an arrow function to a function expression for compatibility.
- Adds and updates unit tests


Testing
======
1. Deploy branch to preview
2. Visit a page on a non-opera mini browser
3. See request in network tab in dev tool via fetch
4. Visit a page on opera mini browser with extreme mode
5. Check ATI to see if page view has tracked

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
